### ISSUE TITLE
another way to parse arguments

### ISSUE THREAD

**Issue by Jaymon** [Jaymon/captain#2]
Found this in my evernote, I think it would be better here so I might actually do it.

---

basically, instead of always having to look at the argparse docs, you could do something like:

``` python
parse_args("--foo=bar [--che] [--baz=[]] [--gat|-g=]")
```

that would basically say foo defaults to bar if not there, che is optional, baz can be a list and is optional, and --get=-g, so both will point to gat, and they are optional and require a value (unlike che which is a boolean value: true if che is there, false if it isn't).

I'm not sure how to handle --to-user, I guess they could be seamlessly changed to to_user so arg.to_user would still work

this could extend the actual argparse library, so you could default to doing things the normal argparse way also

this has no way to define help docs, which sucks, so you couldn't say, che, pass this in to enable che module.

---

Basically, I think I can just have another decorator that allows this, and so help docs would be done the typical captain way, I think this would be pretty cool, not sure it's needed between the other 2 options available but something to think about

**Comment by Jaymon** [comment]
Just using docopt would be better

https://github.com/docopt/docopt

**Comment by Jaymon** [comment]
Something that would be nice to have is a method to actually parse a full comand line string, so passing something like:

```
parse_arguments("--foo=bar --che "1 2 3 4" -b 5 -b 6")
```

would return something like:

```
{
    "foo": ["bar"],
    "che": ["1 2 3 4"],
    "b": ["5", "6"],
}
```

**Comment by Jaymon** [comment]
I'm pretty happy with the argument parsing captain does. There is the simple parsing that happens by just setting up the handle method signature and there is the more complex configuration that can happen through the argument decorators.

Also, I moved the unknown parsing into `datatypes.parse.ArgParser` and it will parse command strings exactly how I commented on Sep 1, 2020. So I'm closing this issue
